include/xvega/grammar/marks.hpp

```diff
@@ -4,473 +4,213 @@
+            {
+                j["width"] = width().value();
+            }
+            if(height().has_value())
+            {
+                j["height"] = height().value();
+            }
+            if(y().has_value())
+            {
+                j["y"] = y().value();
+            }
+            if(y2().has_value())
+            {
+                j["y2"] = y2().value();
+            }
+            if(xOffset().has_value())
+            {
+                j["xOffset"] = xOffset().value();
+            }
+            if(x2Offset().has_value())
+            {
+                j["x2Offset"] = x2Offset().value();
+            }
+            if(yOffset().has_value())
+            {
+                j["yOffset"] = yOffset().value();
+            }
+            if(y2Offset().has_value())
+            {
+                j["y2Offset"] = y2Offset().value();
+            }
+
+            // Fill in Color Properties
+            if(filled().has_value())
+            {
+                j["filled"] = filled().value();
+            }
+            if(color().has_value())
+            {
+                j["color"] = color().value();
+            }
+            if(fill().has_value())
+            {
+                j["fill"] = fill().value();
+            }
+            if(stroke().has_value())
+            {
+                j["stroke"] = stroke().value();
+            }
+            if(blend().has_value())
+            {
+                j["blend"] = blend().value();
+            }
+            if(opacity().has_value())
+            {
+                j["opacity"] = opacity().value();
+            }
+            if(fillOpacity().has_value())
+            {
+                j["fillOpacity"] = fillOpacity().value();
+            }
+            if(strokeOpacity().has_value())
+            {
+                j["strokeOpacity"] = strokeOpacity().value();
+            }
+            if(strokeCap().has_value())
+            {
+                j["strokeCap"] = strokeCap().value();
+            }
+            if(strokeDash().has_value())
+            {
+                j["strokeDash"] = strokeDash().value();
+            }
+            if(strokeDashOffset().has_value())
+            {
+                j["strokeDashOffset"] = strokeDashOffset().value();
+            }
+            if(strokeJoin().has_value())
+            {
+                j["strokeJoin"] = strokeJoin().value();
+            }
+            if(strokeMiterLimit().has_value())
+            {
+                j["strokeMiterLimit"] = strokeMiterLimit().value();
+            }
+            if(strokeWidth().has_value())
+            {
+                j["strokeWidth"] = strokeWidth().value();
+            }
+
+            // Fill in Hyperlink Properties
+            if(href().has_value())
+            {
+                j["href"] = href().value();
+            }
+            if(cursor().has_value())
+            {
+                j["cursor"] = cursor().value();
+            }
+        }
```

Comment:
since `xmarks`is an intermediate class in the CRTP hierarchy, you probably want to prevent it from being instantiated in client code wihle you want to keep it instantiable form inheriting class. A way to achieve this is to declare the constructors, destructor, and assignment operators as protected:

```cpp
template <class D>
struct marks : public xp::xobserved<D>
{
// same as above
protected:
    marks() = default;
    ~marks() = default();

    marks(const marks&) = default;
    mark& operator=(const marks&) = default,

    marks(marks&&) = default;
    marks& operator=(marks&&) = default;
};
```